MD/md-20/README.md

```diff
@@ -0,0 +1,85 @@
+
+### 2. Safe and Secure Access to L2 $MOVE Pool
+
+- **Justification**: The Bridge Relayer keys should not have access to the $MOVE pool. If they did, they could arbitrarily mint L2 $MOVE, effectively offsetting the balance between L1 ERC20 $MOVE and L2 $MOVE, potentially incurring huge losses for Movement Labs.
+
+### 3. Gas Sponsorship for the L2 Swap
+
+- **User Journey**: After a transfer from L1 to L2 is completed, the user expects to receive L2 $MOVE equivalent to the ERC20 $MOVE that was transferred into their Movement Wallet.
+- **Justification**: At the initial step of the swap (when the transfer is completed on Movement L2), the user requires some L2 $MOVE to execute the swap transaction.
+
+### 4. Fallback Mechanism for Failed Swaps
+
+- **User Journey**: The user is waiting for their transfer to complete. There is an AWS outage, and our relayer goes down. This happens after the transfer is completed, but the Swap TX was never called.
+- **Justification**: The swap could fail for several reasons, such as connectivity or liveness issues. In such cases, a suitable mechanism should be implemented to retry or refund the swap.
+
+## Guidance and Suggestions
+
+At the earliest stage of the L1 to L2 transfer, the Bridge Relayer service should transfer some of its owned L2 $MOVE to the user associated with the recipient on the `bridgeTransferId`, **IF** the user does not already possess L2 $MOVE. This is the sponsored transaction, and it should be a very nominal amount.
+
+The safest and most secure way to implement the swap step of the Transfer sequence is to maintain the pattern already implemented in the atomic bridge contracts. Namely, the bridge counterparty module mints an `FA` and gives it to the user.
+
+Currently, in the `atomic_bridge_counterparty.move` module, the FA asset is given to the user, and that is the end of the Transfer. This is where the Swap step occurs.
+
+### Example Cases
+
+#### L1 -> L2
+
+1. The user transfers 10 L1 ERC20 $MOVE.
+2. Upon calling `lock_bridge_transfer` on `atomic_bridge_counterparty.move`, 0.42 L2 $MOVE is deposited _from_ the Bridge Relayer Account _to_ the User Account.
+3. The transfer is initiated, and the lock and secret reveal steps are completed.
+4. `complete_bridge_transfer` is called on `atomic_bridge_counterparty.move` by the Bridge Relayer service.
+5. The user now has 10 BRIDGE-FA tokens and 0.42 L2 $MOVE.
+6. A new function `swap_asset` on `atomic_bridge_counterparty.move` is called:
+    - The `swap_asset` swaps the 10 BRIDGE-FA tokens for L2 $MOVE, paying for the transaction with the user's 0.42 L2 $MOVE.
+    - The `swap_asset` must make a call to _an_external_pool_ of L2 $MOVE.
```

Comment:
Could you please provide more details on how `swap_asset` makes the swap the external pool? Seeing as the risk of operator keys being compromised is the main reason for this change, it seems to me that how this swap happens is a critical part of the design. How are swaps authorized? Who controls the L2 $MOVE in the pool?

Reply:
Good question and I think this is the crux of the MD. The pool would be on another module also owned and deployed by movement labs (but different keys). This can be deployed separately the bridge and the bridge has no need to "own" it, it just needs to be able to call it, similar to how there is a WETH contract for the ETH swap. 

My making it owned by a separate keypair we reduce bridge relayer trust assumptions, was the idea. The module will be very simple.